src/components/dropdownSearch/DropdownSearch.tsx

```diff
@@ -0,0 +1,326 @@
+  onKeyDownDropdownButton?: (keyCode: number) => void;
+  onMouseEnterDropdown?: () => void;
+}
+
+export interface IDropdownSearchProps extends IDropdownSearchOwnProps, IDropdownSearchStateProps, IDropdownSearchDispatchProps { }
+
+export class DropdownSearch extends React.Component<IDropdownSearchProps, void> {
+  filterInput: HTMLDivElement;
+  ulElement: HTMLElement;
+  private textIndex = 0;
+  private dropdownButton: HTMLElement;
+
+  static defaultProps: Partial<IDropdownSearchProps> = {
+    isOpened: false,
+    highlightThreshold: 100,
+    highlightAllFilterResult: false,
+    noResultText: 'No results',
+  };
+
+  private getSelectedOption(): JSX.Element {
+    return <span key={this.props.selectedOption.value}
+                 className='dropdown-selected-value'
+                 data-value={this.props.selectedOption.value}>
+      {this.props.selectedOption.displayValue || this.props.selectedOption.value}
+    </span>;
+  }
+
+  private getDropdownOptions(): JSX.Element[] {
+    const options = _.chain(this.props.options)
+      .filter(
+        (option: IDropdownOption) => {
+          const value = option.displayValue || option.value;
+          return _.isEmpty(this.props.filterText) || s.contains(value.toLowerCase(), this.props.filterText.toLowerCase());
+        },
+      )
+      .map((option: IDropdownOption, index: number, options: IDropdownOption[]) => {
+        const optionClasses = classNames({
+          'state-selected': option.value === this.props.selectedOption.value,
+        });
+        const liClasses = classNames({
+          'active': JSON.stringify(option) === JSON.stringify(this.props.activeOption),
+        });
+
+        const value = option.displayValue || option.value;
+        const valueToShow = !!this.props.highlightAllFilterResult || options.length <= this.props.highlightThreshold
+          ? this.getTextFiltered(value)
+          : value;
+
+        return <li key={option.value}
+                   className={liClasses}>
+          <span className={optionClasses}
+                onMouseDown={(e: React.MouseEvent<HTMLSpanElement>) => this.handleOnOptionClick(e)} data-value={option.value}>
+            {this.getDropdownPrepend(option)}
+            {this.getSvg(option)}
+            {valueToShow}
+          </span>
+        </li>;
+      })
+      .value();
+
+    if (!options.length) {
+      return [
+        <li key='noResultDropdownSearch'>
+          <span>{this.props.noResultText}</span>
+        </li>,
+      ];
+    }
+
+    return options;
+  }
+
+  private getSvg(option: IDropdownOption): JSX.Element {
+    if (option.svg) {
+      return <span key={option.svg.name}
+                   className='value-icon'>
+        <Svg {...option.svg} />
+      </span>;
+    }
+    return null;
+  }
+
+  private getTextFiltered(text: string): (JSX.Element | string)[] | string {
+    this.textIndex = 0;
+    const originalText = new String(text).toString();
```

Comment:
used for the key on element, its the way to create a new string, not a reference